### PR TITLE
Add a --fix option to yarn optimize-images

### PR DESCRIPTION
**Pull request by fnune**
We could consider adding this to CI if we gain confidence in it.

`yarn optimize-images --help` now prints:

```
~/Development/memfault-docs => yarn optimize-images --help
yarn run v1.22.11
$ node ./optimize-images.js --help

  This tool produces minified copies with -min appended to their names
  before the extension and optionally fixes documents that use the
  unoptimized version of images (see '--fix').

  If you run with '--fix', make sure to do it on a clean branch in case
  of erroneous updates. The tool just runs 'sed' under the hood.

  USAGE:

      yarn optimize-images [OPTIONS]

  OPTIONS:

      -r, --reprocess Images for which a -min version already exists will
                      be skipped. To force reprocessing these images, pass
                      the --reprocess option.
      -f, --fix       After processing, runs 'sed' to update existing links
                      to unoptimized images in place.
      -h, --help      Show this help message and exit.

Done in 0.10s.
```

### CODE REVIEW

File: optimize-images.js [hunk 1]
```diff
@@ -111,44 +110,73 @@ if (help) {
...
+
+        console.log(`Total size before: ${formatSize(sizeBefore)}`);
+        console.log(`Total size after:  ${formatSize(sizeAfter)}`);
+        console.log(`Space saved:       ${formatSize(savedTotal)}`);
+
+        const fix =
+            process.argv.includes("--fix") || process.argv.includes("-f");
+        if (fix) {
+            const dir = pt.dirname(require.main.filename);
+            FILETYPES.forEach((filetype) => {
```
[the remark this answers is not in the document]
Comment (noahp) [reply optimize-images.js:164]: sometimes easier to just _avoid sed whenever possible_ :sweat_smile:

**Review by noahp — COMMENTED** [review]
love it! super NIT but might be nice to split the `--fix` option addition + tool change into one commit and the running of it into a second commit, totally good with squash too though

**Review by noahp — APPROVED** [review]

File: optimize-images.js [hunk 2]
```diff
@@ -111,44 +110,73 @@ if (help) {
...
+                const sedExpression = `/${MIN_SUFFIX}.${filetype}/! s/\\.${filetype}/${MIN_SUFFIX}.${filetype}/g`;
```
Comment (noahp) [inline optimize-images.js:165]: I verified prettier won't split image links, so this shouldn't conflict with that.

**Comment by HBehrens** [comment]
I still don't think that optimizations like these should not involve any manual steps and would like to propose again a step when producing the final build that produces both variants and picks the smaller file to be stored as the file all pages refer to.

**Comment by fnune** [comment]
:+1: I'll look up a Docusaurus plugin or otherwise we can look for one of the existing Webpack plugins that help do that.

I don't remember exactly how I started building this script, but I probably checked out some existing solutions with Webpack and couldn't find anything satisfactory. I know Gatsby does this pretty well, so maybe I should see how they do it.

**Pull request merged**